projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-history/replenishment-order-history.component.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Component,
+  OnDestroy,
+  ChangeDetectionStrategy,
+  ElementRef,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  ReplenishmentOrder,
+  ReplenishmentOrderList,
+  RoutingService,
+  TranslationService,
+  UserReplenishmentOrderService,
+} from '@spartacus/core';
+import { ReplenishmentOrderCancellationLaunchDialogService } from '../replenishment-order-details/replenishment-order-cancellation/replenishment-order-cancellation-launch-dialog.service';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { map, take, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-replenishment-order-history',
+  templateUrl: './replenishment-order-history.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ReplenishmentOrderHistoryComponent implements OnDestroy {
+  @ViewChild('element') element: ElementRef;
+
+  private subscription = new Subscription();
+
+  constructor(
+    private routing: RoutingService,
+    private userReplenishmentOrderService: UserReplenishmentOrderService,
+    protected replenishmentOrderCancellationLaunchDialogService: ReplenishmentOrderCancellationLaunchDialogService,
+    private translation: TranslationService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  private PAGE_SIZE = 5;
```

Comment:
please moves these variables with values already 'set' on top of the constructor as convention.

and everything below the constructor only has the functions